Modules/System Tests/Time Zone Management/src/TimeZoneManagementTest.Codeunit.al

```diff
@@ -0,0 +1,332 @@
+
+        // [GIVEN] Time zone that does not have daylight saving time
+        TimeZoneId := CreateTimeZoneThatDoesNotHaveDaylightSavingTime();
+        // [GIVEN] DateTime inside daylight saving time period
+        DateTimeToTest := CreateDateTimeInsideDaylightSavingTimePeriod();
+
+        // [WHEN] Check whether datetime falls within DST for the time zone
+        DateTimeIsInDST := CheckWhetherDateTimeFallsWithinDSTForTheTimeZone(DateTimeToTest, TimeZoneId);
+
+        // [THEN] DateTime shows as not being in DST
+        VerifyDateTimeShowsAsNotBeingInDST(DateTimeIsInDST);
+    end;
+
+    var
+        PermissionsMock: Codeunit "Permissions Mock";
+        DateTimeOffset: Codeunit "DateTime Offset";
+        DaylightSavingTimeInfo: Codeunit "Daylight Saving Time Info";
+        LibraryAssert: Codeunit "Library Assert";
+        IsInitialized: Boolean;
+
+    local procedure Initialize()
+    begin
+        if IsInitialized then
+            exit;
+
+        PermissionsMock.Set('TimeZoneMgt-Objects');
+
+        IsInitialized := true;
+        Commit();
+    end;
+
+    local procedure CreateTimeZoneThatHasDaylightSavingTime(): Text
+    begin
+        exit(GetTimeZoneIdForDSTTimeZone());
+    end;
+
+    local procedure CreateTimeZoneThatDoesNotHaveDaylightSavingTime(): Text
+    begin
+        exit(GetTimeZoneIdForNonDSTTimeZone());
+    end;
+
+    local procedure CreateDateTimeOutsideDaylightSavingTimePeriod(): DateTime
+    begin
+        exit(CreateDateTime(DMY2Date(26, 12, 2022), 140000T));
+    end;
+
+    local procedure CreateDateTimeInsideDaylightSavingTimePeriod(): DateTime
+    begin
+        exit(CreateDateTime(DMY2Date(2, 6, 2022), 140000T));
+    end;
+
+    local procedure RequestUTCOffset(DateTimeToTest: DateTime; TimeZoneId: Text): Duration
+    begin
+        exit(DateTimeOffset.GetUtcOffset(DateTimeToTest, TimeZoneId));
+    end;
+
+    local procedure VerifyOffsetIsCorrectForSelectedTimeZone(Offset: Duration; ExpectedOffsetInHours: Decimal)
+    var
+        ExpectedOffset: Duration;
+        OffsetMismatchErr: Label 'Offset did not match expected offset.', Locked = true;
+    begin
+        ExpectedOffset := ExpectedOffsetInHours * GetHourConversionFactor();
+        LibraryAssert.AreEqual(ExpectedOffset, Offset, OffsetMismatchErr);
+    end;
+
+    local procedure VerifyOffsetBetweenTimeZonesIsCorrect(Offset: Duration)
+    var
+        OffsetMismatchErr: Label 'Offset between time zones did not match expected offset.', Locked = true;
+        ExpectedOffset: Duration;
+    begin
+        ExpectedOffset := -9 * GetHourConversionFactor();
+        LibraryAssert.AreEqual(ExpectedOffset, Offset, OffsetMismatchErr);
+    end;
+
+    local procedure CheckWhetherTimeZoneSupportsDaylightSavingTime(TimeZoneId: Text): Boolean
+    begin
+        exit(DaylightSavingTimeInfo.TimeZoneSupportsDaylightSavingTime(TimeZoneId));
+    end;
+
+    local procedure VerifyDaylightSavingTimeIsNotSupported(SupportsDST: Boolean)
```

Comment:
These local helper methods seem only the called once in the tests - why have them in the first place?

Reply:
It keeps the test procedures better structured in the given-when-then pattern, thus increasing readability.